.github/actions/TestApplicationManifestConsistency/action.ps1

```diff
@@ -0,0 +1,100 @@
+function Test-ApplicationManifestConsistency
+{
+    param(
+        [Parameter(Mandatory = $false)]
+        [string]$ExpectedAppVersion = (GetApplicationVersion),
+        [Parameter(Mandatory = $false)]
+        [string]$ExpectedPlatformVersion = (GetPlatformVersion)
+    )
+    $errors = @()
+
+    $bcAppsManifests = (Get-ChildItem "src" -Recurse -Include app.json).FullName
+    foreach($path in $bcAppsManifests)
+    {
+        $manifest = Get-Content $path | ConvertFrom-Json
+        if ($ExpectedAppVersion -ne $manifest.version)
+        {
+            $errors += "ERROR: Wrong application version in manifest $path. Expected: $ExpectedAppVersion. Actual: $($manifest.version)"
+        }
+        foreach($item in $manifest.dependencies)
+        {
+            if ($ExpectedAppVersion -ne $item.version)
+            {
+                $errors += "ERROR: Wrong dependency version ($($item.name)) in manifest $path. Expected: $ExpectedAppVersion. Actual: $($item.version)"
+            }
+        }
+        if ($manifest.platform -and ($ExpectedPlatformVersion -ne $manifest.platform))
+        {
+            $errors += "ERROR: Wrong platform version in manifest $path. Expected: $ExpectedPlatformVersion. Actual: $($manifest.platform)"
+        }
+        if ($manifest.publisher -ne "Microsoft")
+        {
+            # Allow Partner publisher if path contains 'Partner Test' and is under App\BCApps
+            if ($path -like "*Partner Test*" -and $manifest.publisher -eq "Partner")
+            {
+                Write-Host "Allowing Partner publisher in $path as it contains 'Partner Test' and is under BCApps."
+            }
+            else
+            {
+                $errors += "ERROR: Wrong publisher in manifest $path. Expected: Microsoft. Actual: $($manifest.publisher)"
+            }
+        }
+    }
+
+    if ($errors)
+    {
+        $errors | Write-Host
+        throw "Found $($errors.Count) error(s) in application manifest versions"
+    }
+}
+
+function GetApplicationVersion
+{
+    Import-Module (Join-Path $PSScriptRoot "..\..\..\build\scripts\EnlistmentHelperFunctions.psm1" -Resolve)
+    $repoVersion = Get-ConfigValue -Key "repoVersion" -ConfigType AL-GO
+    return "$repoVersion.0.0"
+}
+
+function GetPlatformVersion
```

Comment:
Too much of an effort, if you ask me.

The _platform_ property indicates a _minumum_ required version, right?

Either check if the same as the app version, or one lower.
Or check if the major version of the _platform_ property is the same as the major version in from the artifact version.